### PR TITLE
feat(fallback): allow falling back w/ overrides

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/playwright-core/src/client/network.ts [hunk 1]
```diff
@@ -142,7 +157,9 @@ export class Request extends ChannelOwner<channels.RequestChannel> implements ap
    * @deprecated
    */
   headers(): Headers {
-    return this._provisionalHeaders.headers();
+    if (this._fallbackOverrides.headers)
```
Comment (yury-s) [inline packages/playwright-core/src/client/network.ts:160]: Remove the `if` block , `return this._fallbackOverrides.headers || this._provisionalHeaders.headers();` should be sufficient

File: packages/playwright-core/types/types.d.ts [hunk 2]
```diff
@@ -14859,8 +14879,44 @@ export interface Route {
...
+     */
+    method?: string;
+
+    /**
+     * If set changes the post data of request
+     */
+    postData?: string|Buffer;
+
+    /**
+     * If set changes the request URL. New URL must have same protocol as original one.
```
Comment (yury-s) [inline packages/playwright-core/types/types.d.ts:14916]: Also explain how next routes will be matched?

File: tests/library/defaultbrowsercontext-2.spec.ts [hunk 3]
```diff
@@ -223,35 +223,3 @@ it('should connect to a browser with the default page', async ({ browserType,cre
   expect(context.pages().length).toBe(1);
   await context.close();
 });
-
-it('route.continue should delete the origin header', async ({ launchPersistent, server, isAndroid, browserName }) => {
```
Comment (yury-s) [inline tests/library/defaultbrowsercontext-2.spec.ts:227]: Please revert this, see corresponding bug, the origin header is not deletable in some browsers.

**Review by yury-s — APPROVED** [review]

File: packages/playwright-core/src/client/network.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/client/network.ts:160]: I need to normalize those (lower case names, etc)

File: packages/playwright-core/types/types.d.ts [hunk 4]
```diff
@@ -14834,8 +14834,28 @@ export interface Route {
...
+   * ```js
+   * await page.route('**\/*', route => {
+   *   // Runs last.
+   *   route.abort();
+   * });
+   * await page.route('**\/*', route => {
+   *   // Runs second.
+   *   route.fallback();
+   * });
+   * await page.route('**\/*', route => {
```
[the remark this answers is not in the document]
Comment (pavelfeldman) [reply packages/playwright-core/types/types.d.ts:14851]: Our generator escapes those...

File: tests/library/defaultbrowsercontext-2.spec.ts (hunk 3 above)
Comment (pavelfeldman) [reply tests/library/defaultbrowsercontext-2.spec.ts:227]: It is a dupe of "should delete the origin header" in tests/page/page-request-continue.spec.ts

**Pull request merged**